### DISCUSSION TITLE
Where should I load a sitemap.xml and a robot.txt ... and can I view it?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37648] [Help] [answered]
So I generated a sitemap from xml-sitempas.com....   good deal, it works with Next...

Now, where can I load it in the file system?   And can I view it? 

Could this be OK?  and can I view on the browser? 

ROOT
| Pages
| Public
  -  Sitemap.xml 
  - robot.txt

**Comment by icyJoseph** [comment, marked as answer]
Yes. They go under the public folder. https://nextjs.org/docs/basic-features/static-file-serving

You can view them at `example.com/Sitemap.xml` or `example.com/robot.txt` as you have them.

**Reply by Kethangkath** [reply]
thanks a bunch man. I placed the file in the root path and had to take a moment of silence in determining whether I chose the correct profession.

**Reply by huykon** [reply]
not work now. your link was died

**Reply by icyJoseph** [reply]
This was originally answered for Pages Router.

In App Router, the recommended way for sitemaps is, https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap

**Comment by Elindo586** [comment]
Ok... so they must be under the ROOT folder and not the the /public folder.

Also.... http://localhost:3000/sitemap.xml   does it work on the localhost? 

I am getting a 404 error with both robot and sitemap

**Reply by icyJoseph** [reply]
No, they must be under public. Next.js makes sure they are available at runtime as if they were at the root.

**Reply by icyJoseph** [reply]
I updated my comment. For some reason on my phone the file structure example you give looks different than on my laptop.

**Comment by Elindo586** [comment]
Oh well... it doesn't work on the localhost, but I can view it live on the browser after upload.
